Big background image cause error when creating a canvas with them

**Describe the bug**
Adding a huge background create a crash when trying to create a canvas with it

**To Reproduce**
Steps to reproduce the behavior:
1. Use !setbackground to set a big image
2. Use !stats or any commands generating canvas and it should crash

**Expected behavior**
This should work normally and display the image instead of a black image
We can either compress the image when saving it or just limit the file size when using !setbackground

**Error message**
```
javax.imageio.IIOException: Error reading PNG image data
        at com.sun.imageio.plugins.png.PNGImageReader.readImage(PNGImageReader.java:1345)
        at com.sun.imageio.plugins.png.PNGImageReader.read(PNGImageReader.java:1614)
        at javax.imageio.ImageIO.read(ImageIO.java:1448)
        at javax.imageio.ImageIO.read(ImageIO.java:1308)
        at utils.Canvas.drawCustomBackground(Canvas.java:130)
        at commands.CommandStats.execute(CommandStats.java:44)
        at core.CommandExecutable.run(CommandExecutable.java:32)
        at commands.CommandManager.<init>(CommandManager.java:14)
        at event.MessageReceived.run(MessageReceived.java:35)
        at core.Bot.onMessageReceived(Bot.java:66)
        at net.dv8tion.jda.api.hooks.ListenerAdapter.onEvent(ListenerAdapter.java:428)
        at net.dv8tion.jda.api.hooks.InterfacedEventManager.handle(InterfacedEventManager.java:96)
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handleInternally(EventManagerProxy.java:82)
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handle(EventManagerProxy.java:69)
        at net.dv8tion.jda.internal.JDAImpl.handleEvent(JDAImpl.java:152)
        at net.dv8tion.jda.internal.handle.MessageCreateHandler.handleInternally(MessageCreateHandler.java:122)
        at net.dv8tion.jda.internal.handle.SocketHandler.handle(SocketHandler.java:36)
        at net.dv8tion.jda.internal.requests.WebSocketClient.onDispatch(WebSocketClient.java:954)
        at net.dv8tion.jda.internal.requests.WebSocketClient.onEvent(WebSocketClient.java:841)
        at net.dv8tion.jda.internal.requests.WebSocketClient.handleEvent(WebSocketClient.java:819)
        at net.dv8tion.jda.internal.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:992)
        at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:385)
        at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:276)
        at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:996)
        at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:755)
        at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108)
        at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64)
        at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45)
Caused by: java.io.EOFException: Unexpected end of ZLIB input stream
        at java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:240)
        at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:158)
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
        at java.io.DataInputStream.readFully(DataInputStream.java:195)
        at com.sun.imageio.plugins.png.PNGImageReader.decodePass(PNGImageReader.java:1119)
        at com.sun.imageio.plugins.png.PNGImageReader.decodeImage(PNGImageReader.java:1223)
        at com.sun.imageio.plugins.png.PNGImageReader.readImage(PNGImageReader.java:1338)```

--- a/config.json
+++ b/config.json
@@ -26,6 +26,7 @@
         "background remove": "Background removed",
         "unlink error": "Impossible to find corresponding user",
         "wrong file type": "The file format is incorrect! Supported format are %format%",
+        "large image": "The file is too large! Please use a file whats under %size%",
         "link": "User linked successfully",
         "background reset": "Your stats background was set back to the default one",
         "unknown player": "Unknown player %player%",

--- a/src/commands/CommandSetBackground.java
+++ b/src/commands/CommandSetBackground.java
@@ -21,6 +21,8 @@ public class CommandSetBackground extends CommandExecutable {
 		if (command.getEvent().getMessage().getAttachments().size() == 0) return (missingFile(this));
 		
 		Attachment file = command.getEvent().getMessage().getAttachments().get(0);
+
+		if(file.getSize() > 8388608) return (fileSizeOverLimit(this, "8MiB"));
 		
 		if (file.getFileExtension().equals("png") || file.getFileExtension().equals("jpg") || file.getFileExtension().equals("jpeg")) {
 			updateBackground(command.getSender().getId(), file);

--- a/src/core/CommandError.java
+++ b/src/core/CommandError.java
@@ -78,6 +78,16 @@ public interface CommandError {
 		MessageSender.wrongFileFormat(cmd.command, format);
 		return (false);
 	}
+
+	/**
+	 * Custom error for too large attachments
+	 * @param format
+	 * @return
+	 */
+	default boolean fileSizeOverLimit(CommandExecutable cmd, String format) {
+		MessageSender.fileSizeOverLimit(cmd.command, format);
+		return (false);
+	}
 	
 	/**
 	 * Custom error for unknown member

--- a/src/utils/MessageSender.java
+++ b/src/utils/MessageSender.java
@@ -71,6 +71,18 @@ public class MessageSender {
 		str += Config.getMessage("wrong file type").replace("%format%", message);
 		command.getEvent().getChannel().sendMessage(str).complete();
 	}
+
+	/**
+	 * Send wrong file size error message
+	 * @param channel
+	 * @param message
+	 */
+	public static void fileSizeOverLimit(Command command, String message) {
+		String str = command.getSender().getAsMention() + ",\n";
+		
+		str += Config.getMessage("large image").replace("%size%", message);
+		command.getEvent().getChannel().sendMessage(str).complete();
+	}
 	
 	/**
 	 * Send unknown player message

